academy/xl-cosmjs/multi-msg.md

```diff
@@ -0,0 +1,496 @@
+---
+title: "Send Multiple Messages"
+order: 3
+description: Send multiple tokens and messages through CosmJs
+tag: deep-dive
+---
+
+# Send Multiple Messages
+
+## Send multiple tokens using `sendTokens`
+
+In the [previous exercise](./first-steps.md), you had Alice send tokens back to the faucet. To refresh your memory, here is the function's signature:
+
+```typescript [https://github.com/cosmos/cosmjs/blob/7aad551/packages/stargate/src/signingstargateclient.ts#L217-L223]
+public async sendTokens(
+    senderAddress: string,
+    recipientAddress: string,
+    amount: readonly Coin[],
+    fee: StdFee | "auto" | number,
+    memo = "",
+): Promise<DeliverTxResponse>;
+```
+
+`Coin[]` is convenient in that Alice can send not just `stake`, but also any number of other coins as long as she owns them. So she can:
+
+<CodeGroup>
+<CodeGroupItem title="Send one token type" active>
+
+```typescript
+const result = await signingClient.sendTokens(
+    alice,
+    faucet,
+    [
+        {
+            denom: "stake",
+            amount: "10000000",
+        },
+    ],
+    "auto",
+)
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="Send two token types">
+
+```typescript
+const result = await signingClient.sendTokens(
+    alice,
+    faucet,
+    [
+        {
+            denom: "stake",
+            amount: "10000000",
+        },
+        {
+            denom: "token",
+            amount: "12",
+        },
+    ],
+    "auto",
+)
+```
+
+</CodeGroupItem>
+</CodeGroup>
+
+However, there are **limitations** here. Alice can only target a single recipient per transaction. If she wants to send tokens to multiple recipients, then she needs to create as many transactions as there are recipients.
+
+Multiple transactions cost a little bit more than packing transfers into the array because of tranaction overhead. The **second limitation** is that separate transfers are not atomic. It's possible that Alice wants to send tokens to two recipients and it's important that that either they both receive them or none of them receive anything.
```

Comment:
```suggestion
Multiple transactions cost a little bit more than packing transfers into the array because of transaction overhead. The **second limitation** is that separate transfers are not atomic. It's possible that Alice wants to send tokens to two recipients and it's important that either they both receive them or none of them receive anything.
```
@rob-Hitchens